### PR TITLE
Fix MSAA for OpenGL ES 3.0 core. (third attempt)

### LINKED ISSUE #3128: WebGL triangle demo has GL_INVALID_FRAMEBUFFER_OPERATION
The web demos all work except the triangle demo, which does not render anything and emits the following in the console:

     glBlitFramebufferCHROMIUM: framebuffer incomplete
     
The triangle demo is the only WebGL demo that enables multisampling.

### PR DESCRIPTION
**Pull request by prideout**
This fixes `GL_FRAMEBUFFER_INCOMPLETE_MULTISAMPLE` caused by the following spec clause being violated:

```
The value of RENDERBUFFER_SAMPLES [must be] the same for all
attached renderbuffers and, if the attached images are a mix of
renderbuffers and textures, the value of RENDERBUFFER_SAMPLES
is zero.
```

Tested on Chrome with Triangle and Helmet demos (after enabling MSAA).
Also tried running the automation suite in a debug build on macOS, to see if any asserts or GL errors would be triggered.

Fixes #3128.

### CODE REVIEW

File: filament/backend/src/opengl/OpenGLDriver.cpp [hunk 1]
```diff
@@ -899,21 +902,43 @@ void OpenGLDriver::framebufferTexture(backend::TargetBufferInfo const& binfo,
         }
     } else
 #endif
-    { // here we emulate ext.EXT_multisampled_render_to_texture
+    if (!any(t->usage & TextureUsage::SAMPLEABLE)) {
```
Comment (pixelflinger) [inline filament/backend/src/opengl/OpenGLDriver.cpp:905]: I think this test is not enough. We also need a sidecar if the target render buffer is not multi-sampled.
That is, if the user is asking for a "magic resolve" into a render buffer.

File: filament/backend/src/opengl/OpenGLDriver.cpp [hunk 2]
```diff
@@ -899,21 +902,43 @@ void OpenGLDriver::framebufferTexture(backend::TargetBufferInfo const& binfo,
...
         gl.bindFramebuffer(GL_FRAMEBUFFER, rt->gl.fbo);
+        glFramebufferRenderbuffer(GL_FRAMEBUFFER, attachment, GL_RENDERBUFFER, t->gl.id);
+
+        // Clear the resolve bit for this particular attachment. Note that other attachment(s)
+        // might be sampleable, so this does not necessarily prevent the resolve from occuring.
+        resolveFlags = TargetBufferFlags::NONE;
 
+    } else {
+        assert(rt->gl.samples > 1);
+        assert(glIsTexture(t->gl.id));
```
Comment (pixelflinger) [inline filament/backend/src/opengl/OpenGLDriver.cpp:922]: per the comment above, this check should be removed.

File: filament/backend/src/opengl/OpenGLDriver.cpp [hunk 3]
```diff
@@ -926,14 +951,8 @@ void OpenGLDriver::framebufferTexture(backend::TargetBufferInfo const& binfo,
...
             case GL_TEXTURE_2D:
-                if (any(t->usage & TextureUsage::SAMPLEABLE)) {
-                    glFramebufferTexture2D(GL_FRAMEBUFFER, attachment,
-                            target, t->gl.id, binfo.level);
-                } else {
-                    assert(target == GL_TEXTURE_2D);
-                    glFramebufferRenderbuffer(GL_FRAMEBUFFER, attachment,
-                            GL_RENDERBUFFER, t->gl.id);
-                }
+                glFramebufferTexture2D(GL_FRAMEBUFFER, attachment,
```
Comment (pixelflinger) [inline filament/backend/src/opengl/OpenGLDriver.cpp:954]: Per the comment above, we can still end-up here with a render buffer

**Review by pixelflinger — CHANGES_REQUESTED** [review]

File: filament/backend/src/opengl/OpenGLDriver.cpp (hunk 1 above)
Comment (prideout) [reply filament/backend/src/opengl/OpenGLDriver.cpp:905]: If the target render buffer is not multi-sampled, then this code is unreachable; see line 850.

Comment (pixelflinger) [reply filament/backend/src/opengl/OpenGLDriver.cpp:905]: No no that's not what I mean. I meant if the render buffer is not multi-sampled, but the render target is. Again, this is the case of a "magic resolve" into a render buffer.

Comment (prideout) [reply filament/backend/src/opengl/OpenGLDriver.cpp]: This is complicated!  Okay I added another commit to make this change, and thankfully WebGL is still working.

**Review by pixelflinger — APPROVED** [review]

**Pull request merged**